examples/tabular_image.py

```diff
@@ -0,0 +1,201 @@
+parser.add_argument("--epochs", type=int, default=30)
+parser.add_argument("--seed", type=int, default=0)
+parser.add_argument(
+    "--model",
+    type=str,
+    default="google/vit-base-patch16-224-in21k",
+    choices=[
+        "resnet18",
+        "google/vit-base-patch16-224-in21k",
+    ],
+)
+
+args = parser.parse_args()
+
+# ResNet-18 - Image Embedded
+# Best Val Acc: 0.2861, Best Test Acc: 0.2799
+# ViT - Image Embedded
+# Best Val Acc: 0.4221, Best Test Acc: 0.4269
+
+
+class ImageToEmbedding:
+    def __init__(self, model: str, device: torch.device):
+        if "resnet" in model:
+            from torchvision import transforms
+            model = torch.hub.load(
+                'pytorch/vision:v0.10.0',
+                model,
+                pretrained=True,
+            )
+            # Remove the last linear layer:
+            self.model = torch.nn.Sequential(
+                *(list(model.children())[:-1])).to(device)
+            # Preprocess transformations:
+            self.preprocess = transforms.Compose([
+                transforms.Resize(256),
+                transforms.CenterCrop(224),
+                transforms.ToTensor(),
+                transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                                     std=[0.229, 0.224, 0.225]),
+            ])
+        elif "vit" in model:
+            from transformers import AutoImageProcessor, ViTModel
+            self.preprocess = AutoImageProcessor.from_pretrained(model)
+            self.model = ViTModel.from_pretrained(model).to(device)
+        self.model.eval()
+        self.device = device
+
+    def forward_retrieve(self, path_to_images: list[str]) -> list[Image]:
+        images: list[Image] = []
+        for path_to_image in path_to_images:
+            try:
+                image = Image.open(path_to_image)
+            except Exception:
+                # There is one image does not exist, thus create a black one:
+                image = Image.new("RGB", (600, 471))
+            images.append(image.copy())
+            image.close()
+        images = [image.convert('RGB') for image in images]
+        return images
+
+    def forward_embed(self, images: list[Image]) -> Tensor:
+        if "ViT" in str(self.preprocess):
+            inputs = self.preprocess(images, return_tensors="pt")
+            inputs["pixel_values"] = inputs["pixel_values"].to(device)
+            with torch.no_grad():
+                res = self.model(**inputs).pooler_output.cpu().detach()
+            return res
+        else:
+            inputs = [self.preprocess(image) for image in images]
+            inputs = torch.stack(inputs, dim=0).to(self.device)
+            with torch.no_grad():
+                res = self.model(inputs).cpu().detach()
+            return res.view(len(images), -1)
+
+    def __call__(self, path_to_images: List[str]) -> Tensor:
+        images = self.forward_retrieve(path_to_images)
+        return self.forward_embed(images)
```

Comment:
Does it make sense to create a parent class so that you don't need to write `forward_retrieve` and `__call__`? You can inherit that class here.

Reply:
The parent class is in PyTorch Frame side. `forward_embed` is all most users need to write.
Also, let's write a careful doc-string for the parent class so that people know how to use it.